package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -0,0 +1,605 @@
+                    except ValueError:
+                        raise ValueError(
+                            "there is no reference attributes"
+                            " (elements, types, or names)"
+                            " in this universe to guess mass from")
+
+        if indices_to_guess is not None:
+            atom_types = atom_types[indices_to_guess]
+
+        masses = np.array([self.get_atom_mass(atom)
+                           for atom in atom_types], dtype=np.float64)
+        return masses
+
+    def get_atom_mass(self, element):
+        """Return the atomic mass in u for *element*.
+        Masses are looked up in :data:`MDAnalysis.guesser.tables.masses`.
+
+        .. Warning:: Untill version 3.0.0 unknown masses are set to 0.0
+
+        """
+        try:
+            return tables.masses[element]
+        except KeyError:
+            try:
+                return tables.masses[element.upper()]
+            except KeyError:
+                warnings.warn(
+                    "Unknown masses are set to 0.0 for current version, "
+                    "this will be deprecated in version 3.0.0 and replaced by"
+                    " Masse's no_value_label (np.nan)",
+                    PendingDeprecationWarning)
+                return 0.0
+
+    def guess_atom_mass(self, atomname):
+        """Guess a mass based on the atom name.
+
+        :func:`guess_atom_element` is used to determine the kind of atom.
+
+    .. warning:: Untill version 3.0.0 anything not recognized is simply
+        set to 0.0; if you rely on the masses you might want to double-check.
+        """
+        return self.get_atom_mass(self.guess_atom_element(atomname))
+
+    def guess_types(self, atom_types=None, masses=None, indices_to_guess=None):
+        """Guess the atom type of many atoms based on atom name
+        Parameters
+        ----------
+        atom_types (optional)
+          atoms names if types guessing is desired to be from names
+        masses (optional)
+           atoms masses if types guessing is desired to be from masses
+        indices_to_guess (optional)
+          Mask array for partially guess types for certain atoms
+
+        Returns
+        -------
+        atom_types : np.ndarray dtype object
+
+        Raises
+        ------
+        :exc:`ValueError` if there is no names
+         or masses to guess types from.
+
+        """
+        if atom_types is None and masses is None:
+            try:
+                atom_types = self._universe.atoms.names
+            except AttributeError:
+                try:
+                    # if names is not available we can guess types from masses
+                    masses = self._universe.atoms.masses
+                except AttributeError:
+                    raise ValueError(
+                        "there is no reference attributes in this universe"
+                        "to guess types from")
+
+        if masses is not None:
```

Comment:
I'm not sure if I got it right... If your point is why we don't check if elements/types info already exists in the universe before guessing them, this step takes place in the ``guess_attr()`` method (plus it handles partial/full guessing options of the attribute).

If you mean why we guess ``atomtypes`` if we have the ``element`` attribute already exists... the reason is this will probably break various topologies's old behavior

Reply:
Apologies I missed this, let me have a longer think about this tomorrow. I'll request a review from myself so I can keep track of it.

Reply:
@aya9aladdin sorry for taking so long with this. @lilyminium and I had a chat regarding this branch and we couldn't work out what use case this branch represents. As far as I can tell `guess_element_from_mass` is completely new and doesn't seem to reproduce any existing Reader behaviour (although I am likely missing something). Could you provide some insights on this?

Should this be brand new behaviour / not necessary for anything, then we would suggest not having this here. Elements from mass is a rather flaky due to issues with mass repartitioning schemes and coarse graining.

Reply:
ahh got you... yes indeed guessing elements from masses is a newly introduced feature (cant remember who suggested it tbh) no it doesn't reproduce old behavior, will remove it